pkg/streamingpromql/operators/deduplicate_and_merge_test.go

```diff
@@ -153,20 +156,82 @@ func TestDeduplicateAndMerge(t *testing.T) {
 
 			expectConflict: true,
 		},
+		"with dropName set partially and runDelayedNameRemoval=false": {
+			inputSeries: []labels.Labels{
+				labels.FromStrings(labels.MetricName, "metric", "foo", "1"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "2"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "3"),
+			},
+			inputData: []types.InstantVectorSeriesData{
+				{Floats: []promql.FPoint{{T: 0, F: 0}, {T: 1, F: 1}}},
+				{Floats: []promql.FPoint{{T: 0, F: 10}, {T: 1, F: 11}}},
+				{Floats: []promql.FPoint{{T: 0, F: 20}, {T: 1, F: 21}}},
+			},
+			inputDropName: []bool{
+				true,
+				false,
+				true,
+			},
+
+			expectedOutputSeries: []labels.Labels{
+				labels.FromStrings(labels.MetricName, "metric", "foo", "1"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "2"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "3"),
+			},
+			expectedOutputData: []types.InstantVectorSeriesData{
+				{Floats: []promql.FPoint{{T: 0, F: 0}, {T: 1, F: 1}}},
+				{Floats: []promql.FPoint{{T: 0, F: 10}, {T: 1, F: 11}}},
+				{Floats: []promql.FPoint{{T: 0, F: 20}, {T: 1, F: 21}}},
+			},
+			expectedOutputDropName: []bool{
+				true,
+				false,
+				true,
+			},
+		},
+		"with dropName set partially and runDelayedNameRemoval=true": {
+			inputSeries: []labels.Labels{
+				labels.FromStrings(labels.MetricName, "metric", "foo", "1"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "2"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "3"),
+			},
+			inputData: []types.InstantVectorSeriesData{
+				{Floats: []promql.FPoint{{T: 0, F: 0}, {T: 1, F: 1}}},
+				{Floats: []promql.FPoint{{T: 0, F: 10}, {T: 1, F: 11}}},
+				{Floats: []promql.FPoint{{T: 0, F: 20}, {T: 1, F: 21}}},
+			},
+			inputDropName: []bool{
+				true,
+				false,
+				true,
+			},
+			runDelayedNameRemoval: true,
+
+			expectedOutputSeries: []labels.Labels{
+				labels.FromStrings("foo", "1"),
+				labels.FromStrings(labels.MetricName, "metric", "foo", "2"),
+				labels.FromStrings("foo", "3"),
+			},
+			expectedOutputData: []types.InstantVectorSeriesData{
+				{Floats: []promql.FPoint{{T: 0, F: 0}, {T: 1, F: 1}}},
+				{Floats: []promql.FPoint{{T: 0, F: 10}, {T: 1, F: 11}}},
+				{Floats: []promql.FPoint{{T: 0, F: 20}, {T: 1, F: 21}}},
+			},
```

Comment:
Might be good to be explicit here and show that we expect `DropName` to be false for all series.